### ISSUE TITLE
Froward of Criterion returns dict, so in `train_one_epoch()` you call loss.backward() on dict which throws an error.

### ISSUE THREAD

**Issue by kristiyan-hristov** [Intellindust-AI-Lab/DEIMv2#103]
In file `engine/solver/clas_engine.py` in the function `train_one_epoch` you have:
```
loss: torch.Tensor = criterion(preds, labels, epoch)

optimizer.zero_grad()
loss.backward()
```

but the forward method of your `DEIMCriterion` returns a dict, which does not have `backward` method. So when I try to run it I get and Error.
